src/Workleap.OpenApi.MSBuild/OasdiffManager.cs

```diff
@@ -1,35 +1,81 @@
         this._httpClientWrapper = httpClientWrapper;
         this._processWrapper = processWrapper;
-        this._toolDirectory = Path.Combine(openApiToolsDirectoryPath, "oasdiff", OasdiffVersion);
+        this._openApiToolsDirectory = openApiToolsDirectoryPath;
+        this._oasdiffDirectory = Path.Combine(openApiToolsDirectoryPath, "oasdiff", OasdiffVersion);
     }
 
+    private bool SkipRun { get; set; }
+    
     public async Task InstallOasdiffAsync(CancellationToken cancellationToken)
     {
-        Directory.CreateDirectory(this._toolDirectory);
+        try
+        {
+            this._loggerWrapper.LogMessage("Starting Oasdiff installation.");
+            
+            Directory.CreateDirectory(this._oasdiffDirectory);
+
+            var oasdiffFileName = GetOasdiffFileName();
+            var url = $"https://github.com/Tufin/oasdiff/releases/download/v{OasdiffVersion}/{oasdiffFileName}";
+
+            await this._httpClientWrapper.DownloadFileToDestinationAsync(url, Path.Combine(this._oasdiffDirectory, oasdiffFileName), cancellationToken);
+            await this.DecompressDownloadedFileAsync(oasdiffFileName, cancellationToken);
+            
+            this._loggerWrapper.LogMessage("Oasdiff installation completed.");
+        }
+        catch (Exception)
+        {
+            this.SkipRun = true;
+            throw;
+        }
+    }
 
-        var oasdiffFileName = GetOasdiffFileName();
-        var url = $"https://github.com/Tufin/oasdiff/releases/download/v{OasdiffVersion}/{oasdiffFileName}";
+    public async Task RunOasdiffAsync(IEnumerable<string> openApiSpecFiles, IEnumerable<string> generatedOpenApiSpecFiles, CancellationToken cancellationToken)
+    {
+        if (this.SkipRun)
+        {
+            return;
+        }
+        
+        var generatedOpenApiSpecFilesList = generatedOpenApiSpecFiles.ToList();
+        var oasdiffExecutePath = Path.Combine(this._oasdiffDirectory, RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "oasdiff.exe" : "oasdiff");
 
-        await this._httpClientWrapper.DownloadFileToDestinationAsync(url, Path.Combine(this._toolDirectory, oasdiffFileName), cancellationToken);
-        await this.DecompressDownloadedFileAsync(oasdiffFileName, cancellationToken);
+        foreach (var specFile in openApiSpecFiles)
+        {
+            this._loggerWrapper.LogMessage($"Starting Oasdiff comparison with {specFile}.");
+            
+            var fileName = Path.GetFileName(specFile);
+            var baseSpecRelativePath = Path.GetRelativePath(this._openApiToolsDirectory, specFile);
+            var newSpecRelativePath = Path.GetRelativePath(this._openApiToolsDirectory, generatedOpenApiSpecFilesList.First(x => x.Contains(fileName)));
+            await this._processWrapper.RunProcessAsync(oasdiffExecutePath, new[] { "diff", baseSpecRelativePath, newSpecRelativePath, "--exclude-elements", "description,examples,title,summary", "-f", "text" }, cancellationToken);
```

Comment:
Did we consider to generate a text report as well? It could help for CI builds as the generated file could be attached to the pipeline run